evalml/tests/component_tests/test_knn_imputer.py

```diff
@@ -0,0 +1,201 @@
+    X_expected_arr = pd.DataFrame(
+        [
+            [1, 0, 1, 2],
+            [1, 2, 3, 2],
+            [10, 2, 5, 2],
+            [10, 2, 5, 2],
+            [6, 2, 7, 0],
+            [1, 2, 3, 4],
+            [1, 2, 3, 2],
+        ],
+    )
+    X_t = transformer.fit_transform(X)
+    assert_frame_equal(X_expected_arr, X_t, check_dtype=False)
+
+
+def test_knn_imputer_2_neighbors():
+    X = pd.DataFrame([[np.nan, 0, 3, np.nan], [1, 2, 3, 2], [1, 2, 3, 0]])
+    # test impute_strategy
+    transformer = KNNImputer(number_neighbors=2)
+    X_expected_arr = pd.DataFrame(
+        [[1, 0, 3, 1], [1, 2, 3, 2], [1, 2, 3, 0]],
+    )
+    X_t = transformer.fit_transform(X)
+    assert_frame_equal(X_expected_arr, X_t, check_dtype=False)
+
+
+@pytest.mark.parametrize("data_type", ["pd", "ww"])
+def test_knn_imputer_all_bool_return_original(data_type, make_data_type):
+    X = pd.DataFrame([True, True, False, True, True], dtype=bool)
+    y = pd.Series([1, 0, 0, 1, 0])
+    X = make_data_type(data_type, X)
+    y = make_data_type(data_type, y)
+    X_expected_arr = pd.DataFrame([True, True, False, True, True], dtype=bool)
+    imputer = KNNImputer()
+    imputer.fit(X, y)
+    X_t = imputer.transform(X)
+    assert_frame_equal(X_expected_arr, X_t)
+
+
+@pytest.mark.parametrize("data_type", ["pd", "ww"])
+def test_knn_imputer_boolean_dtype(data_type, make_data_type):
+    X = pd.DataFrame(
+        [True, np.nan, False, np.nan, True],
+        [True, True, False, True, True],
+    )
+    X.ww.init(logical_types={0: "BooleanNullable"})
+    y = pd.Series([1, 0, 0, 1, 0])
+    X_expected_arr = pd.DataFrame(
+        [True, True, False, True, True],
+        [True, True, False, True, True],
+        dtype="boolean",
+    )
+    X = make_data_type(data_type, X)
+    imputer = KNNImputer(number_neighbors=1)
+    X_t = imputer.fit_transform(X, y)
+    assert_frame_equal(X_expected_arr, X_t)
+
+
+@pytest.mark.parametrize("data_type", ["pd", "ww"])
+def test_knn_imputer_multitype_with_one_bool(data_type, make_data_type):
+    X_multi = pd.DataFrame(
+        {
+            "bool with nan": pd.Series([True, np.nan, False, np.nan, False]),
+            "bool no nan": pd.Series([False, False, False, False, True], dtype=bool),
+        },
+    )
+    X_multi.ww.init(logical_types={"bool with nan": "BooleanNullable"})
+    y = pd.Series([1, 0, 0, 1, 0])
+    X_multi_expected_arr = pd.DataFrame(
+        {
+            "bool with nan": pd.Series(
+                [True, True, False, True, False],
+                dtype="boolean",
+            ),
+            "bool no nan": pd.Series([False, False, False, False, True], dtype=bool),
+        },
+    )
+    X_multi = make_data_type(data_type, X_multi)
+
+    imputer = KNNImputer(number_neighbors=1)
+    imputer.fit(X_multi, y)
+    X_multi_t = imputer.transform(X_multi)
+    assert_frame_equal(X_multi_expected_arr, X_multi_t)
+
+
+def test_knn_imputer_all_bool():
```

Comment:
This test is almost functionally identical to `test_knn_imputer_all_bool_return_original`, we don't need both